functional-tests/CloudbaseInit.Tests.ps1

```diff
@@ -230,9 +230,23 @@ function after.cloudbaseinit.plugins.windows.winrmcertificateauth.ConfigWinRMCer
 }
 function before.cloudbaseinit.plugins.common.networkconfig.NetworkConfigPlugin {
 
+    It "a network adapter should exist" {
+        {
+            $netName = $global:CLOUDBASE_INIT_NET_NAME
+            Rename-NetAdapter -Name $global:CLOUDBASE_INIT_NET_NAME -NewName "not_cbs_init0" -ErrorAction SilentlyContinue
+        } | Should -Not -Throw
+    }
+
 }
 function after.cloudbaseinit.plugins.common.networkconfig.NetworkConfigPlugin {
-
+    It "a network adapter should exist and have a proper name" {
+        {
+            Get-NetAdapter -Name $global:CLOUDBASE_INIT_NET_NAME
+            if (!("10.196.59.2" -in (Get-NetIPAddress -InterfaceAlias cbs_init_eth0).IPAddress)) {

```

Comment:
Should we define a constant for this address? Does the address itself have any meaning? Should we use something like 19.19.19.2 to avoid a network that's actually being used?

Reply:
This address is hardcoded on the metadata network-interfaces file. The netmask / network is just a working dummy one, as it will not interfere with anything, as the TAP adapter is disconnected and even if you set a gateway/static adddress, Windows will not consider it.

Reply:
This IP address, as it is going to be used in other metadata services' testing, can be though made a constant at the top, like with the network name, it s probably a good tradeoff.